src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -129,29 +129,59 @@ public void OnResponseHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
                         throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
                     }
 
-                    if (name.SequenceEqual(s_statusHeaderName))
+                    if (name[0] == 58) // ':'
                     {
                         if (_state == StreamState.ExpectingTrailingHeaders)
                         {
                             // Pseudo-headers not allowed in trailers.
                             if (NetEventSource.IsEnabled) _connection.Trace("Pseudo-header in trailer headers.");
                             throw new HttpRequestException(SR.net_http_invalid_response);
                         }
-
-                        if (value.Length != 3)
-                            throw new Exception("Invalid status code");
-
-                        // Copied from HttpConnection
-                        byte status1 = value[0], status2 = value[1], status3 = value[2];
-                        if (!IsDigit(status1) || !IsDigit(status2) || !IsDigit(status3))
+                        if (name.SequenceEqual(s_statusHeaderName))
+                        {
+                            if (_response.StatusCode != 0)
+                            {
+                                if (NetEventSource.IsEnabled) _connection.Trace("Received duplicate status headers.");
+                                throw new HttpRequestException(SR.net_http_invalid_response);
+                            }
+
+                            if (value.Length != 3)
+                            {
+                                if (NetEventSource.IsEnabled) _connection.Trace($"Received status headers '{System.Text.Encoding.ASCII.GetString(value)}'.");
+                                throw new HttpRequestException(SR.net_http_invalid_response);
+                            }
+
+                            // Copied from HttpConnection
+                            byte status1 = value[0], status2 = value[1], status3 = value[2];
```

Comment:
I know you copied this code from elsewhere. But defining multiple local variables on the same line is against typical CoreFx coding practices.  It decreases readability of the code.

Reply:
updated.